### PR TITLE
feat: export grid selection as csv

### PR DESCRIPTION
**Pull request by ddimaria**
- [x] Select rectangle on grid, select Download Selection as CSV, downloads CSV with grid file name and the file name.
- [x] CSV export in Command Palette
- [x] CSV export in DataMenu
- [x] Add Shortcut Command + Shift + E (for export, follows Figma export command)
- [x] Add Rust tests

### CODE REVIEW

File: src/grid/controller/Grid.ts [hunk 1]
```diff
@@ -604,6 +604,24 @@ export class Grid {
 
   //#endregion
 
+  //#region Exports
+
+  exportCsvSelection(): string {
+    if (!this.gridController) throw new Error('Expected grid to be defined in Grid');
+
+    debugTimeReset();
+    console.log(rectangleToRect(sheets.sheet.cursor.getRectangle()));
```
Comment (davidkircos) [inline src/grid/controller/Grid.ts:613]: remove log

**Review by davidkircos — APPROVED** [review]

File: src/grid/actions/downloadSelectionAsCSV.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { downloadFile } from '@/helpers/downloadFileInBrowser';
+import { grid } from '../controller/Grid';
+
+export const downloadSelectionAsCSV = (fileName: string) => {
+  const csv = grid.exportCsvSelection();
+  downloadFile(fileName, csv, 'text/plain', 'csv');
+};
```
Comment (jimniels) [inline src/grid/actions/downloadSelectionAsCSV.ts:7]: We have been slowly working on a centralized place for defining a bunch of actions/labels that get re-used like this.  (We haven't done it yet for "Copy selection as PNG" but should...)

It's probably worth sticking this in there and re-using it?

`src/actions.ts` is the file

Something like

```js
// src/actions.ts
export const downloadSelectionAsCSV = {
  label: 'Download selection as CSV",
  run: (fileName: string) => {
    const csv = grid.exportCsvSelection();
    downloadFile(fileName, csv, 'text/plain', 'csv');
  }
};
```

Comment (jimniels) [reply src/grid/actions/downloadSelectionAsCSV.ts:7]: Also minor casing thing:

We have it named `exportCsvSelection` in another file but `downloadSelectionAsCSV` here. We should either capitalize "CSV" everywhere, or stick to camelCase.

I would say we stick to camelCase, as we'll do things like `fileId` elsewhere, so this should be `downloadSelectionAsCsv`

File: src/ui/menus/CommandPalette/ListItems/Edit.tsx [hunk 3]
```diff
@@ -109,6 +111,24 @@ const ListItems = [
...
+          action={() => {
+            downloadSelectionAsCSV(name);
+          }}
+          icon={<Download />}
+          shortcut="E"
+          shortcutModifiers={[KeyboardSymbols.Command, KeyboardSymbols.Shift]}
+        />
+      );
+    },
+  },
```
Comment (jimniels) [inline src/ui/menus/CommandPalette/ListItems/Edit.tsx:131]: Per the comment about centralizing things in `src/actions.ts` here you could import the label/action and re-use it.

File: src/ui/menus/TopBar/SubMenus/DataMenu.tsx [hunk 4]
```diff
@@ -32,13 +30,19 @@ export const DataMenu = () => {
...
+        <MenuItem
+          onClick={() => {
+            downloadSelectionAsCSV(fileName);
+          }}
+        >
+          <MenuLineItem
+            primary="Selection to CSV"
+            Icon={UploadFile}
+            secondary={KeyboardSymbols.Command + KeyboardSymbols.Shift + 'E'}
+          />
```
Comment (jimniels) [inline src/ui/menus/TopBar/SubMenus/DataMenu.tsx:43]: We're using a different icon in this menu than the other menus.

CleanShot 2023-11-20 at 15 44 30@2x

We should use the same icon we're using in the other menus

CleanShot 2023-11-20 at 15 44 36@2x

CleanShot 2023-11-20 at 15 44 45@2x

**Review by jimniels — CHANGES_REQUESTED** [review]
Looks pretty good. Love this feature! Just a few things to cleanup then it LGTM

File: src/ui/menus/TopBar/SubMenus/DataMenu.tsx (hunk 4 above)
Comment (jimniels) [reply src/ui/menus/TopBar/SubMenus/DataMenu.tsx:43]: @davidkircos can you update the icon on this one to use the same one we're using in the command palette? Looks like it's `<Download />`

**Review by jimniels — APPROVED** [review]
One thing (update icon) and then LGTM 👍

**Pull request merged**